torchci/components/MultiSelectPicker.tsx

```diff
@@ -0,0 +1,114 @@
+// a component that has a list of options and allows the user to select multiple options, with a 'select all' and 'clear' button.
+
+import {
+  Button,
+  Checkbox,
+  FormControl,
+  InputLabel,
+  MenuItem,
+  Select,
+  SelectChangeEvent,
+} from "@mui/material";
+import { useState } from "react";
+
+export default function MultiSelectPicker({
+  selected,
+  setSelected,
+  options,
+  label,
+  renderValue,
+  style,
+}: {
+  selected: string[] | undefined;
+  setSelected: any;
+  options: string[];
+  label: string;
+  renderValue: (selected: string[]) => string;
+  style?: { [key: string]: any };
+}) {
+  var [selectedItems, setSelectedItems] = useState(selected ?? []);
```

Comment:
Why have both selected and selectedItems?

Reply:
'selected' is the initial selection, 'setselected' is the method you pass as callback 'onchange', and selectedItems is the internal state. Will update the naming to make it less confusing